src/ray/core_worker/core_worker.cc

```diff
@@ -2323,32 +2336,61 @@ void CoreWorker::HandleWaitForActorOutOfScope(
   actor_manager_->WaitForActorOutOfScope(actor_id, std::move(respond));
 }
 
-void CoreWorker::HandleWaitForObjectEviction(
-    const rpc::WaitForObjectEvictionRequest &request,
-    rpc::WaitForObjectEvictionReply *reply, rpc::SendReplyCallback send_reply_callback) {
+void CoreWorker::HandleSubscribeForObjectEviction(
+    const rpc::SubscribeForObjectEvictionRequest &request,
+    rpc::SubscribeForObjectEvictionReply *reply,
+    rpc::SendReplyCallback send_reply_callback) {
   // TODO(swang): Drop requests from raylets that executed an older version of
   // the task.
   if (HandleWrongRecipient(WorkerID::FromBinary(request.intended_worker_id()),
                            send_reply_callback)) {
     return;
   }
 
+  const auto subscriber_node_id =
+      NodeID::FromBinary(request.subscriber_address().raylet_id());
   // Send a response to trigger unpinning the object when it is no longer in scope.
-  auto respond = [send_reply_callback](const ObjectID &object_id) {
-    RAY_LOG(DEBUG) << "Replying to HandleWaitForObjectEviction for " << object_id;
-    send_reply_callback(Status::OK(), nullptr, nullptr);
+  auto respond = [this, subscriber_node_id](const ObjectID &object_id) {
+    RAY_LOG(DEBUG) << "Object id: " << object_id << " is deleted. Unpinning the object.";
```

Comment:
Why is that? We use this type of pattern everywhere though (e.g., gcs server). I will change the log regardless, but I'd like to know reasoning?

Reply:
Ah, you said "we tend to use". I guess there's just no consensus here then.

Reply:
Hmm the GCS server uses the pattern that you have above? I've seen the one that I recommended consistently throughout the core worker and raylet. We should really standardize on some log record styles. 😂

Reply:
Yeah there's no common style. I think the ideal solution here is to just stop using stream-based logging and make it structured logging. (so that it will be consistent by the logging framework). It was on the roadmap, but since I stopped working on logging now, there's no progress..